src/Protocol/ActivityPub/Receiver.php

```diff
@@ -1890,14 +1890,20 @@ private static function processObject(array $object, string $actor)
 		$object_data['receiver_urls']  = self::getReceiverURL($object);
 		$object_data['receiver']       = $receivers;
 		$object_data['reception_type'] = $reception_types;
+		$object_data['unlisted']       = false;
 
 		if (!empty($object['pixelfed:capabilities'])) {
 			$object_data['capabilities'] = self::getCapabilities($object);
 		}
 
-		$object_data['unlisted'] = in_array(-1, $object_data['receiver']);
-		unset($object_data['receiver'][-1]);
-		unset($object_data['reception_type'][-1]);
+		if (in_array(-1, $object_data['receiver'])) {
+			$object_data['unlisted'] = true;
+			unset($object_data['receiver'][-1]);
+
+			if (array_key_exists(-1, $object_data['reception_type'])) {
```

Comment:
I thought you simplified to a single condition on line 1899, why the change of heart?

Reply:
Sorry, totally forgot about this: I updated to [PHPStan 2.0](https://phpstan.org/blog/phpstan-2-0-released-level-10-elephpants) and now PHPStan started complaining about this line too. :see_no_evil: 

I'll try to refactor the method again.

Reply:
I've refactored the whole method to check for `-1` as key in advanced. This way a value with this key will never been set and don't need to be `unset()` again.